aten/src/ATen/native/SoftMax.cpp

```diff
@@ -65,7 +64,7 @@ TORCH_META_FUNC(_softmax_backward_data)
   auto grad_input_options =
       grad.options().memory_format(LEGACY_CONTIGUOUS_MEMORY_FORMAT);
 
-  bool half_to_float = grad.scalar_type() != input.scalar_type();
+  bool half_to_float = grad.scalar_type() != output.scalar_type();
```

Comment:
hm, no, in the half-> float  case `output` will be float, `grad` will be float too, but you'd need to produce `half` gradInput.